src/renderers/WebGLRenderer.js

```diff
@@ -1821,6 +1821,10 @@ THREE.WebGLRenderer = function ( parameters ) {
 
 			return a.object.renderOrder - b.object.renderOrder;
 
+		} else if ( !!a.material.program && !!b.material.program && a.material.program.id !== b.material.program.id ) {
```

Comment:
Well... if you are refactoring... async shader compiling? https://twitter.com/Tojiro/status/584820688772431872